apps/editor.planx.uk/src/pages/FlowEditor/data/types.ts

```diff
@@ -46,5 +47,10 @@ export const fromSlug = (slug: string): TYPES | undefined => {
   const entry = Object.entries(SLUGS).find(
     ([_typeString, currentSlug]) => currentSlug === slug,
   );
+
+  // TextInput and EnhancedTextInput share a value in order to keep 
+  // the Select component consistent
+  if (entry?.[1] === "enhanced-text-input") return TYPES.TextInput;
```

Comment:
This is the hacky sort of thing I was hoping to avoid by having a totally separate component, but it doesn't feel _too_ bad currently...?

When we redesign the component selection we'll need to keep in mind that these are no longer 1:1 mappings.